packages/scripts/src/findCentroid.ts

```diff
@@ -0,0 +1,19 @@
+import { strict as assert } from "assert";
+
+/**
+  Finds the centroid (mean) of a set of n-dimensional vectors.
+ */
+export const findCentroid = (vectors: number[][]) => {
+  assert(vectors.length > 0, "Cannot find centroid of 0 vectors!");
+  assert(
+    vectors.find((vector) => vectors[0].length !== vector.length) === undefined,
+    "Vector dimensionality mismatch!"
+  );
+  return (
+    vectors
```

Comment:
there are some libs to help w common linear algebra ops like this that'd make the code little more readable than these raw array operations. 

i worked w a Professional Service eng on something similar a few months back. can't find the lib he used, but asked him..will get back w it, and we can see if applicable here

Reply:
It's done... and it's in the scripts.

Looked for n-dimensional vector libraries but didn't find much of quality. Obviously 2-4D is well-trodden ground.

Reply:
> It's done... and it's in the scripts.

what do you mean?

Reply:
1. Let's not worry *too* much about the legibility/maintainability of the scripts package
2. I already implemented and tested the minimal vector math I need, so I'm less enthused to swap it for some library. As I mentioned, I looked for libraries but good n-dimensional ones were not jumping out to me

Reply:
ok i got the library, it's TensorFlow.js - https://js.tensorflow.org/api/4.15.0/ 

it actually uses hardware acceleration, so theoretically runs faster/more efficiently than standard array ops in addition to having a clearer API for these operations. 

for example, here's the ChatGPT-assisted version of findCentroid with TF.js:

```ts
export async function findCentroid(vectors: number[][]) {
  const meanTensor = tf.tensor(vectors).mean(0);
  return await centroidTensor.array();
}
```

Reply:
How strongly do you feel about this? Will Kanopy or wherever this runs noticeably benefit from (let alone be able to take advantage of) hardware acceleration? I don't really feel strongly at all, except that my implementation is done, works, is tested, is pretty straightforward tbh, is fit for purpose, is fast enough on my computer, etc, so swapping it for another library now smacks of premature optimization and goldplating...

Reply:
i also don't feel strongly here. hardware acceleration aspect might matter when number of queries we're processing increases dramatically, but that can be a future problem. 

re kanopy taking advantage of the hardware acceleration, i believe it would optimize CPU use with tensorflow c++ bindings but not GPU acceleration

personally, i find using tensorflow clearer to read and advocate it for that reason if nothing else. 

and less direct to this PR, i think it's valuable for our team to start familiarizing ourselves with ML libraries and techniques like tensorflow as we dive deeper into the AI/ML space.

not going to block the PR for this.

Reply:
Just for the sake of completeness, I tried your code out, but it doesn't work. I wanted to avoid going down a rabbit hole with a new library, but the docs say mean() is a reduction function that collapses a vector into a scalar. I tried with the keepDims parameter, but it doesn't work. I looked for another function that would find the centroid, but couldn't find one. This is why I dislike the attitude that one should never do their own math when a library exists. There is such a thing as overkill.